torch/nn/grad.py

```diff
@@ -101,29 +59,11 @@ def conv1d_weight(input, weight_size, grad_output, stride=1, padding=0, dilation
         >>> F.grad.conv1d_weight(input, weight.shape, grad_output)
 
     """
-    stride = _single(stride)
-    padding = _single(padding)
-    dilation = _single(dilation)
-    in_channels = input.shape[1]
-    out_channels = grad_output.shape[1]
-    min_batch = input.shape[0]
-
-    grad_output = grad_output.contiguous().repeat(1, in_channels // groups, 1)
-    grad_output = grad_output.contiguous().view(
-        grad_output.shape[0] * grad_output.shape[1], 1, grad_output.shape[2])
-
-    input = input.contiguous().view(1, input.shape[0] * input.shape[1],
-                                    input.shape[2])
+    weight = grad_output.new_empty(weight_size)
```

Comment:
here and below you can make this more efficient by not allocating a full Tensor:
```python
weight = grad_putput.new_empty(1).expand(weight_size)

```

Reply:
Thanks, this is because we will only be reading from `weight` tensor and `expand` creates a view with the right shape instead of actually allocating it?

Reply:
Exactly, since we never use the content anyways, we can make it a single random value instead of allocating a full size Tensor of random values.